### PR TITLE
geo/geomfn: implement ST_TransScale({geometry,float8,float8,float8,float8})

### LINKED ISSUE #49061: geo/geomfn: implement ST_TransScale({geometry,float8,float8,float8,float8})
Implement `ST_TransScale` on arguments {geometry,float8,float8,float8,float8}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_TransScale.html).

*Observers: Please react to this issue if you need this functionality.*

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end (your exhaustive unit tests go the above mentioned packages!).
* Ensure the documentation is regenerated by calling `make buildshort`. You can also play with it by calling `./cockroach demo --empty` afterwards.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

The following additional guidance has been issued on implementing this function:

implement ST_Affine and use

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-09-01T15:16:16Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by adzimzf**
Implement ST_TransScale in CRDB

Fixes #49061.

Release note (sql change): Implemented the geometry-based builtins
`ST_TransScale({geometry,float8,float8,float8,float8})`

### CODE REVIEW

File: pkg/geo/geomfn/affine_transforms.go [hunk 1]
```diff
@@ -281,3 +281,13 @@ func RotateWithXY(g geo.Geometry, rotRadians, x, y float64) (geo.Geometry, error
...
 }
+
+// TransScale returns a modified Geometry whose coordinates are
+// translate by delX and delY and scale by xFactor and yFactor
+func TransScale(g geo.Geometry, delX, delY, xFactor, yFactor float64) (geo.Geometry, error) {
+	g, err := Translate(g, []float64{delX, delY})
+	if err != nil {
+		return g, err
+	}
+	return Scale(g, []float64{xFactor, yFactor})
```
Comment (otan) [inline pkg/geo/geomfn/affine_transforms.go:292]: we should actually speed to one operation by using the `affine` transformation:

```

return affine(
--
  | t,
  AffineMatrix([][]float64{
   {xFactor, 0, 0, xFactor * deltaX},
   {0, yFactor, 0, yFactor * deltaY},
   {0, 0, 1, 0},
   {0, 0, 0, 1},
   }),
   )


```

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 2]
```diff
@@ -5599,6 +5599,33 @@ See http://developers.google.com/maps/documentation/utilities/polylinealgorithm`
 			},
 		},
 	),
+	"st_transscale": makeBuiltin(defProps(),
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"geometry", types.Geometry},
+				{"deltaX", types.Float},
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:5606]: these should be camel case, i.e. `delta_x`, `delta_y`, `x_factor`, `y_factor`

File: pkg/geo/geomfn/affine_transforms.go [hunk 3]
```diff
@@ -281,3 +281,13 @@ func RotateWithXY(g geo.Geometry, rotRadians, x, y float64) (geo.Geometry, error
 		},
 	)
 }
+
+// TransScale returns a modified Geometry whose coordinates are
+// translate by delX and delY and scale by xFactor and yFactor
+func TransScale(g geo.Geometry, delX, delY, xFactor, yFactor float64) (geo.Geometry, error) {
```
Comment (otan) [inline pkg/geo/geomfn/affine_transforms.go:287]: nit: can you rename delX and delY to deltaX/deltaY

File: pkg/geo/geomfn/affine_transforms.go (hunk 1 above)
Comment (adzimzf) [reply pkg/geo/geomfn/affine_transforms.go:292]: I've tried to google it how to just use only using an affine matrix, but not luck

File: pkg/sql/sem/builtins/geo_builtins.go (hunk 2 above)
Comment (adzimzf) [reply pkg/sql/sem/builtins/geo_builtins.go:5606]: noted

File: pkg/geo/geomfn/affine_transforms.go (hunk 3 above)
Comment (adzimzf) [reply pkg/geo/geomfn/affine_transforms.go:287]: agree, that's clearer

**Comment by adzimzf** [comment]
Hi @otan  I have some updates

**Comment by otan** [comment]
you'll need to regenerate the generated files again (`make buildshort`)

**Comment by adzimzf** [comment]
> you'll need to regenerate the generated files again (`make buildshort`)

Sorry forgot to run `make buildshort`,
updated now

**Review by otan — APPROVED** [review]
bors r+

thanks!!!

**Pull request merged**